### PR TITLE
DOXYGEN: Changes to stream.h and system.h

### PR DESCRIPTION
**Pull request by b-gent**
Review and improve doxygen comments in stream.h and system.h.

Current HTML output can be found here:
https://b-gent.github.io/osystem_groups/d5/d81/group__common__system__module.html

### CODE REVIEW

File: common/system.h
Comment (mgerhardy) [inline common/system.h:251]: Looks like there is a comment in a comment here. Some compilers will complain about that.

Comment (b-gent) [reply common/system.h:251]: corrected, thanks

File: common/stream.h
Comment (criezy) [inline common/stream.h:773]: I am commenting on the first relevant case, but this applies to all the comments in this class.
Here the idea is to read a value using the stream endianness and return it in native endianness. So for example if we create the stream with `bigEndian = true`, then `readUint16()` does the same thing as `readUint16BE()` (that is inherited from `ReadStream`): it reads a 16 bit unsigned integer big endian value and convert it to the native endianness.

Here are some details
The code here is confusing, and I think it was meant to use
```C++
return (_bigEndian) ? FROM_BE_16(val) : FROM_LE_16(val);
```

The behaviour of `TO_BE_16` is exactly the same as `FROM_BE_16`: if we are on a big endian system they both do nothing, and if we are on a little endian system they both swap the bytes.

But semantically the current implementation seems to say "I am reading from a stream that uses native endianness and I want a big endian value". And it would make more sense to say "I am reading from a big endian stream and I want the value in native endianness.

That is because often the streams are the game data file, and we know the endianness of those files (usually this is the endianness of the system on which the game was released, so for example BE for mac games and LE for DOS games). And we read those and want to convert them to the native endianness.

And you can see this logic in the ReadStream implementation. For example:
```C++
uint16 readUint16BE() {
	uint16 val;
	read(&val, 2);
	return FROM_BE_16(val);
}
```
I think here it was meant to do the same, and in actual fact the original implementation was:
```C++
inline uint16 readUint16() {
	return (_bigEndian) ? readUint16BE() : readUint16LE();
}
```
This was changed in a big refactoring to the current code, and I believe the use of `TO_*` instead of `FROM_*` was a mistake. I will change those, but maybe wait until after this PR is merged to avoid conflicts.

**Review by criezy — CHANGES_REQUESTED** [review]
Good work on `stream.h`!
There are some changes to make though (see the various comments).

I will review `system.h` tomorrow.

**Comment by b-gent** [comment]
Thanks for taking this up so quickly @criezy , I'll start working on the comments during the day tomorrow.

File: common/system.h
Comment (criezy) [inline common/system.h:1478]: As a warning, there is currently another pull request open (PR #2588) to use `Common::U32String` instead of `const char*`. We will need to revisit that doc when it is merged (since then the `ISO LATIN 1` will no longer be correct as it will use UTF-32 encoding instead).

**Review by criezy — COMMENTED** [review]
And I have also reviewed the changes for the `system.h` file now.

**Comment by b-gent** [comment]
@sev-  you might want to review if you have time

**Comment by criezy** [comment]
One more thing: for the `OSystem` class, I think it would be good to have the various `get*Manager()` functions in the `Module slots` group rather than `Miscellaneous`.

And maybe also rename `Module slots` to `Subsystem modules`. The word `slot` doesn't really speak to me.

File: common/stream.h
Comment (b-gent) [reply common/stream.h:108]: Yes, I like the 'client code' as it prevents the confusion that 'calling code' may bring, just like I did here.

File: common/stream.h
Comment (b-gent) [reply common/stream.h:122]: I thought so as well but then decided it was maybe written like this on purpose. Changing to 'Obtain the current value of the stream position indicator.'.

File: common/stream.h
Comment (b-gent) [reply common/stream.h:307]: Thanks for the explanation, replaced in other occurrences as well.

File: common/stream.h
Comment (b-gent) [reply common/stream.h:636]: Changed to ' Read a string in Pascal format, (..)' hope that's fine

File: common/stream.h
Comment (b-gent) [reply common/stream.h:715]: typo

File: common/stream.h
Comment (b-gent) [reply common/stream.h:799]: fully agree, I overlooked it

File: common/stream.h
Comment (b-gent) [reply common/stream.h:773]: Ok, so right now I'm changing this (and other occurrences respectively) to:
``
	 * Read an unsigned 16-bit word using the stream endianness
	 * and return it in native endianness.
``
and then this can be changed when you make the fix.
Nice to see we can also make some improvements in the code as part of this project.

**Comment by b-gent** [comment]
> One more thing: for the `OSystem` class, I think it would be good to have the various `get*Manager()` functions in the `Module slots` group rather than `Miscellaneous`.
> 
> And maybe also rename `Module slots` to `Subsystem modules`. The word `slot` doesn't really speak to me.

Looks like this right now:
https://b-gent.github.io/osystem_groups/d5/d81/group__common__system__module.html

I assumed getFilesystemFactory() should also be moved there, let me know if that's correct.

**Comment by b-gent** [comment]
All comments are fixed now. Would be great to merge asap to avoid conflicts @criezy

**Comment by b-gent** [comment]
Retry DeepCode

File: common/stream.h
Comment (criezy) [inline common/stream.h:697]: This should be `SEEK_CUR`, not `SEK_CUR` (sorry that was a typo in my comment).
I will still merge and then fix the typo in tree.

**Review by criezy — APPROVED** [review]

**Comment by criezy** [comment]
Merged manually to fix the conflict (there was one already!) and fix the typo on `SEK_CUR`.

**Comment by b-gent** [comment]
Thanks!

**Pull request closed without merging**
